pallets/auctions/src/lib.rs

```diff
@@ -596,3 +620,107 @@ impl<T: Config> NftAuction<T::AccountId, T::AuctionId, BalanceOf<T>, Auction<T>>
 		Ok(())
 	}
 }
+
+///
+/// Implementation of TopUpAuction
+///
+impl<T: Config> NftAuction<T::AccountId, T::AuctionId, BalanceOf<T>, Auction<T>> for TopUpAuction<T> {
+	///
+	/// Places a bid on an TopUpAuction
+	///
+	/// the same functionality as bidding on English auction
+	///
+	/// Also registers individual bids for the final settlement
+	///
+	fn bid(&mut self, bidder: T::AccountId, value: BalanceOf<T>) -> DispatchResult {
+		// Lock / Unlock funds
+		<T as crate::Config>::Currency::set_lock(AUCTION_LOCK_ID, &bidder, value, WithdrawReasons::all());
+
+		self.general_data.last_bid = Some((bidder.clone(), value));
+		// Set next minimal bid
+		let bid_step = Permill::from_percent(<T as crate::Config>::BidStepPerc::get()).mul_floor(value);
+		self.general_data.next_bid_min = value.checked_add(&bid_step).ok_or(Error::<T>::BidOverflow)?;
+
+		let current_block = <frame_system::Pallet<T>>::block_number();
+
+		self.specific_data.bids.push(Bid {
+			who: bidder,
+			when: current_block,
```

Comment:
`timestamp` or `block_nr`? not rly sure about the conventions :)

Reply:
definitely not timestamp :) i think i might remove it completely, not useful for TopUp, will be used in candle i suppose